src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -0,0 +1,209 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <vector>
+
+#include "Domain/Creators/DomainCreator.hpp"  // IWYU pragma: keep
+#include "Domain/Domain.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace domain {
+namespace creators {
+
+/*!
+ * \ingroup DomainCreatorsGroup
+ *
+ * \brief A Domain for testing purposes.
+ *
+ * \image html binary_compact_object_domain.png "A BHNS domain."
+ *
+ * Creates a 3D Domain consisting of four/five layers of blocks:
+ * - 0: The block at the center of each compact object, if not excised.
+ * - 1: The first layer, has the blocks for each compact object.
+ * - 2: The second layer, has the blocks that surround each object with a cube.
+ * - 3: The third layer, has the blocks that surround each cube with a
+ * half-cube.
+ * - 4: The fourth layer is made of the blocks that form the outer sphere.
+ *
+ * `ObjectA` and `ObjectB` refer to the Shell or Sphere making up each compact
+ * object.
+ * `enveloping cube` and `enveloping sphere` refer to the bounding surfaces of
+ * the outer Sphere of ten wedges making up the last layer of the domain.
+ *
+ * The plane along which the domain divides into two hemispheres is referred
+ * to as the "cutting plane".
```

Comment:
Please add to the doxygen the following information about the coordinates:
- the outer sphere is centered at the origin
- the enveloping cube is centered at the origin
- In the final coordinates, the cutting plane always intersects the x-axis at the origin